### PR TITLE
feat(signer): Test environment

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
We need a test environment that includes not just the signer itself, but also the cycles ledger and cycles depositor.

# Changes
- Define a test environment including the signer, multiple canisters, and utilities for funding a user.
  - Note: This is largely taken from the `papi` test environment.

# Tests
This environment is used in tests on a feature branch, but those tests will be merged separately to limit PR size.

### CODE REVIEW

File: src/signer/canister/tests/it/utils/test_environment.rs [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+            Principal::from_text("dmvof-2tilt-3xmvh-c7tbj-n3whk-k2i6b-2s2ge-xoo3d-wjuw3-ijpuw-eae")
+                .unwrap(),
+            Principal::from_text("kjerd-nj73t-u3hhp-jcj4d-g7w56-qlrvb-gguta-45yve-336zs-sunxa-zqe")
+                .unwrap(),
+            Principal::from_text("zxhav-yshtx-vhzs2-nvuu3-jrq66-bidn2-put3y-ulwcf-2gb2o-ykfco-sae")
+                .unwrap(),
+            Principal::from_text("nggqm-p5ozz-i5hfv-bejmq-2gtow-4dtqw-vjatn-4b4yw-s5mzs-i46su-6ae")
+                .unwrap(),
+        ];
+        let unauthorized_user =
```
Comment (lmuntaner) [inline src/signer/canister/tests/it/utils/test_environment.rs:99]: Unauthorized for what?

I see this is a controller of the cycles depositor. Is that related to being unauthozied?

File: src/signer/canister/tests/it/utils/test_environment.rs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        };
+        ans.fund_user(Self::USER_INITIAL_BALANCE);
+        ans
+    }
+}
+impl TestSetup {
+    /// The user's initial balance.
+    pub const USER_INITIAL_BALANCE: u128 = 100_000_000_000;
+    /// Deposit 100 * the ledger fee in the user's ledger wallet. That should be enough to be getting on with.
+    pub fn fund_user(&self, megasquigs: u128) {
```
Comment (lmuntaner) [inline src/signer/canister/tests/it/utils/test_environment.rs:132]: What is a `megasquigs`?

File: src/signer/canister/tests/it/utils/test_environment.rs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            cycles_depositor,
+        };
+        ans.fund_user(Self::USER_INITIAL_BALANCE);
+        ans
+    }
+}
+impl TestSetup {
+    /// The user's initial balance.
+    pub const USER_INITIAL_BALANCE: u128 = 100_000_000_000;
+    /// Deposit 100 * the ledger fee in the user's ledger wallet. That should be enough to be getting on with.
```
Comment (lmuntaner) [inline src/signer/canister/tests/it/utils/test_environment.rs:131]: Is this comment related to the constant or `fund_user`?

**Review by lmuntaner — APPROVED** [review]
It looks very nice!

Approving, only a couple of comments.

File: src/signer/canister/tests/it/utils/test_environment.rs (hunk 1 above)
Comment (bitdivine) [reply src/signer/canister/tests/it/utils/test_environment.rs:99]: This is the user I use whenever I need an external bad guy in a test.  Like Eve the evil persona in a communication between Alice and Bob.  Random third parties should not be able to steal resources or break th eprotocol.

File: src/signer/canister/tests/it/utils/test_environment.rs (hunk 3 above)
Comment (bitdivine) [reply src/signer/canister/tests/it/utils/test_environment.rs:131]: Good catch; the fixed deposit has been replaced by a variable amount.

File: src/signer/canister/tests/it/utils/test_environment.rs (hunk 2 above)
Comment (bitdivine) [reply src/signer/canister/tests/it/utils/test_environment.rs:132]: A British colloquial term for "lots of money".  I'll replace it with something like `cycles`.

File: src/signer/canister/tests/it/utils/test_environment.rs (hunk 1 above)
Comment (bitdivine) [reply src/signer/canister/tests/it/utils/test_environment.rs:99]: Added a comment to that effect.

File: src/signer/canister/tests/it/utils/test_environment.rs (hunk 3 above)
Comment (bitdivine) [reply src/signer/canister/tests/it/utils/test_environment.rs]: Fixed the comment.  Thanks!

File: src/signer/canister/tests/it/utils/test_environment.rs (hunk 2 above)
Comment (bitdivine) [reply src/signer/canister/tests/it/utils/test_environment.rs]: Renamed.
